force-app/main/releaseGating/classes/ReleaseGateAPIService.cls

```diff
@@ -105,16 +104,31 @@ public virtual with sharing class ReleaseGateAPIService {
      * @return           A GetReleaseGateActionResultModel
      */
     public virtual GetReleaseGateActionResultModel getReleaseGates() {
-        Object result = this.makeCall(
-            GET_ACTION,
-            new Map<String, Object>{ VERSION_PARAMETER => this.releaseGateRegistry.apiVersion }
-        );
-        GetReleaseGateActionResultModel resultModel = (GetReleaseGateActionResultModel) JSON.deserialize(
-            (String) result,
-            GetReleaseGateActionResultModel.class
-        );
-        this.verifyCallableResult(resultModel, GET_ACTION);
-        return resultModel;
+        //Customize application perm is required for all release gating actions
+        this.locateUserPermissionService().checkCustomizeApplicationForCurrentUser();
+        try {
+            Object result = this.makeCall(
+                GET_ACTION,
+                new Map<String, Object>{ VERSION_PARAMETER => this.releaseGateRegistry.apiVersion }
+            );
+            GetReleaseGateActionResultModel resultModel = (GetReleaseGateActionResultModel) JSON.deserialize(
+                (String) result,
+                GetReleaseGateActionResultModel.class
+            );
+            return resultModel;
+        } catch (System.JSONException je) {
+            String errorMessage = String.format(
+                Label.stgErrorSettingsProductInvalidResponse,
+                new List<Object>{ this.releaseGateRegistry.name }
+            );
+            UTIL_Debug.debug(System.LoggingLevel.ERROR, je.getMessage());
```

Comment:
I think we could remove this debugging line

Reply:
This is one of those locations that I feel we may get value out of. CCE has requested that we incorporate more UTIL_Debug than less within our code, as it's only viewable if the debug logs are turned on and accessible within the package.

Reply:
Yes, here and below debug statements were added for Support/CCE to help with debugging

Reply:
oh great! good to know, thanks for explaining it